### PR TITLE
PyGEOS to 0.9, GEOS to 3.9.0, Add Py39, Drop Py35, MSVC to 2019

### PR DESCRIPTION
**Pull request by caspervdw**
Quite some changes here, I hope everything will pass.

One thing I am not sure of is the OSX builds. I removed the explicit `osx_image: xcode10.1` setting for the Py38 build, basically following the advice on "relying on defaults"  from https://github.com/matthew-brett/multibuild#macos. But I really don' t know much about MacOS so I am in the dark here what is best to distribute.

Do you have an opinion about this @mwtoews ?

Ref. https://github.com/pygeos/pygeos/issues/250

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -83,8 +76,8 @@ build_script:
   - pip install wheel numpy
```
Comment (jorisvandenbossche) [inline appveyor.yml]:
```suggestion
  - pip install wheel numpy cython
```

(that's the reason appveyor once are failing right now)

**Comment by caspervdw** [comment]
Linux builds are passing!

Windows builds are suffering from a bug with the GEOS 3.9.0 build/test:
- https://trac.osgeo.org/geos/ticket/1081
- (patch) https://git.osgeo.org/gitea/geos/geos/pulls/113

I think it is best to skip the tests. We have sufficient of them in pygeos (and we run them) anyway.

All OSX builds are failing with the following log:
```
Warning: You are using macOS 10.13.
We (and Apple) do not provide support for this old version.
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Twitter or any other official channels. You are responsible for resolving
any issues you experience while you are running this
old version.

Error: ccache: no bottle available!
You can try to install from source with:
  brew install --build-from-source ccache

Please note building from source is unsupported. You will encounter build
failures with some formulae. If you experience any issues please create pull
requests instead of asking for help on Homebrew's GitHub, Twitter or any other
official channels.
```

Not sure how to deal with that. I'll bump multibuild to the latest version, maybe it is fixed upstream.

File: appveyor.yml [hunk 2]
```diff
@@ -1,37 +1,30 @@
 ---
-image: Visual Studio 2017
+image: Visual Studio 2019
```
Comment (jorisvandenbossche) [inline appveyor.yml:2]: Is it OK to bump this? 
(no idea if it is a problem, to be clear)

**Comment by jorisvandenbossche** [comment]
Seems to be passing now!

File: appveyor.yml (hunk 2 above)
Comment (caspervdw) [reply appveyor.yml:2]: It seems that 2017 and 2019 are both using C++ 14.x so they are all fine to use with Py35-39

Comment (mwtoews) [reply appveyor.yml:2]: See [this up-to-date resource for CPython](https://wiki.python.org/moin/WindowsCompilers), which confirms this is a suitable change.

**Comment by vincentsarago** [comment]
@caspervdw looking at https://pypi.org/project/pygeos/#files only the wheels for windows have been deployed. is this expected?

**Comment by caspervdw** [comment]
Thanks for the headsup! I think that the travis builds where skipped because `[skip travis]` was hidden in the description of the merge commit. Apparently appveyor has different rules for this, because `[skip appveyor]` was there also.

I triggered the travis build so Linux and OSX wheels should be up in an hour.

**Pull request merged**